transformer_engine/common/normalization/common.h

```diff
@@ -139,16 +140,28 @@ enum class NVTE_Norm_Backend { Te, Cudnn };
 enum class NVTE_Norm_Type { LayerNorm, RMSNorm };
 enum class NVTE_Norm_Stage { Forward, Backward };
 
-constexpr uint64_t get_key(NVTE_Norm_Type NormType, NVTE_Norm_Stage NormStage, DType wtype,
-                           DType itype, DType otype, DType ctype, uint64_t batch_size,
-                           uint64_t hidden_size, bool zero_centered_gamma, bool is_tuned);
+using TupleKeyType = std::tuple<uint64_t, uint64_t, uint64_t, bool>;
```

Comment:
Using a tuple as a key instead of `int64_t` is a big improvement. I'd lobby for going even further and adding more tuple entries so we can entirely avoid weird bitwise manipulation with `general_key`.